File: client/src/types/pool.ts
export interface Pool {
  poolId: number;
  name: string;
  address: string;
  latitude: number;
  longitude: number;
  poolType: 'Indoor' | 'Outdoor';
  lengthMeters: number | null;
  laneCount: number | null;
  isAccessible: boolean;
  phone: string | null;
  website: string | null;
}

export interface ScheduleResult {
  scheduleId: number;
  swimType: string;
  dayOfWeek: number;
  startTime: string;
  endTime: string;
}

export interface ScoreBreakdown {
  proximity: number;
  poolLength: number;
  laneCount: number;
  scheduleConvenience: number;
}

export interface PoolSearchResult extends Pool {
  distanceKm: number;
  compositeScore: number;
  scores: ScoreBreakdown;
  matchingSchedules: ScheduleResult[];
}

export const SWIM_TYPES = [
  'Lane Swim',
  'Leisure Swim',
  'Aquafit',
  'Women Only',
  'Older Adult',
  'Family',
] as const;

export type SwimType = (typeof SWIM_TYPES)[number];

export const DAY_NAMES = ['Sun', 'Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat'] as const;
export const DAY_NAMES_FULL = ['Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday'] as const;

File: src/Api/Models/Pool.cs
namespace Api.Models;

public class Pool
{
    public int Id { get; set; }
    public required string Name { get; set; }
    public required string Address { get; set; }
    public double Latitude { get; set; }
    public double Longitude { get; set; }
    public required string PoolType { get; set; } // "Indoor", "Outdoor", or "Both"
    public double? LengthMeters { get; set; }
    public int? LaneCount { get; set; }
    public bool IsAccessible { get; set; }
    public string? Phone { get; set; }
    public string? Website { get; set; }
    public string? ImageUrl { get; set; }
    public string? AmenitiesJson { get; set; } // JSON array: ["Water Slide", "Sauna", ...]
    public int? TorontoLocationId { get; set; }
    public bool IsActive { get; set; } = true;

    public List<Schedule> Schedules { get; set; } = [];
}
